File: app/src/main/java/com/example/projectironplate/database/habitGoals/HabitGoals.java
package com.example.projectironplate.database.habitGoals;


/**
 * a tabe containing the goals for the associated habits
 */
public class HabitGoals {
    private int id;
    private HabitType habitType;
    private double goal;

    /**
     * constructor for the HabitGoals class
     * @param id id of habit
     * @param habit name of habit
     * @param goal goal of habit
     */
    public HabitGoals (int id, String habit, double goal) {
        this.id = id;
        this.habitType = HabitType.fromDbValue(habit);
        this.goal = goal;
    }

    /**
     * collection of getters
     */
    public int getId() { return id; }
    public HabitType getHabit() { return habitType; }
    public double getGoal() { return goal; }

    /**
     * collection of setters
     */
    public void setId(int Id) { this.id = Id; }
    public void setGoal(double goal) { this.goal = goal; }

    @Override
    public String toString() {
        return "HabitGoals{" +
                "rowId=" + id +
                "habit=" + habitType +
                "goal=" + goal +
                '}';
    }
}

File: app/src/main/java/com/example/projectironplate/database/habitGoals/HabitGoals.kt
package com.example.projectironplate.database.habitGoals

import com.example.projectironplate.database.habitGoals.HabitType.Companion.fromDbValue


/**
 * a tabe containing the goals for the associated habits
 */
class HabitGoals(private var id: Int, habit: String?, private var goal: Double) {
    private val habitType: HabitType?

    /**
     * constructor for the HabitGoals class
     * @param id id of habit
     * @param habit name of habit
     * @param goal goal of habit
     */
    init {
        this.habitType = fromDbValue(habit)
    }

    /**
     * collection of getters
     */
    fun getId(): Int {
        return id
    }

    fun getHabit(): HabitType? {
        return habitType
    }

    fun getGoal(): Double {
        return goal
    }

    /**
     * collection of setters
     */
    fun setId(Id: Int) {
        this.id = Id
    }

    fun setGoal(goal: Double) {
        this.goal = goal
    }

    override fun toString(): String {
        return "HabitGoals{" +
                "rowId=" + id +
                "habit=" + habitType +
                "goal=" + goal +
                '}'
    }
}
